### PR TITLE
fix: move msi to be optional

### PR DESCRIPTION
**Pull request by junjiezhang1997**
**What type of PR is this?**
/kind fix

**What this PR does / why we need it**:
Move msi to be optional.
**Which issue(s) this PR fixes**:
#3810 

Fixes #

**Requirements**:
- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
Worked like a charm :)
/approve
/lgtm

File: vhdbuilder/packer/init-variables.sh [hunk 1]
```diff
@@ -203,7 +203,12 @@ WINDOWS_IMAGE_URL=""
 windows_servercore_image_url=""
 windows_nanoserver_image_url=""
 windows_private_packages_url=""
-windows_msi_resource_string=${WINDOWS_MSI_RESOURCE_STRING}
+
+windows_msi_resource_string=()
```
Comment (ShiqianTao) [inline vhdbuilder/packer/init-variables.sh:207]: Why an empty string is `()` instead of `""`?

Comment (junjiezhang1997) [reply vhdbuilder/packer/init-variables.sh:207]: This variable is an array so the initialization is ().

Comment (junjiezhang1997) [reply vhdbuilder/packer/init-variables.sh:207]: Maybe I should rename it to avoid confusion,

Comment (AbelHu) [reply vhdbuilder/packer/init-variables.sh:207]: Or you can construct the array in line 209 since we only use one msi for now.

Comment (junjiezhang1997) [reply vhdbuilder/packer/init-variables.sh:209]: The condition could set msi as an optional choice for creating windows vm, which could help solve the issue mentioned in pr description.

Comment (junjiezhang1997) [reply vhdbuilder/packer/init-variables.sh]: Also rename this variable to windows_msi_resource_strings to avoid confusion.

**Review by AbelHu — APPROVED** [review]
LGTM. Thanks for the PR.

**Comment by mjudeikis** [comment]
Thanks for the quick turnaround! OSS for the win :)

**Pull request merged**